### PR TITLE
added split_array function (WIP)

### PR DESCRIPTION
**Pull request by haihongren**
- added split_array function to split an array that has nested arrays
- updated functions.md with example.

### CODE REVIEW

**Comment by ardias** [comment]
@haihongren Is this something that could not be accomplished with other functions? For example `split_objects`?
Also, please add a test similar to what we have added recently for other functions.
Thank you

**Comment by Kav91** [comment]
@ardias split_objects cannot do whats required, have been discussing this scenario with @haihongren .
I'm syncing up with him tomorrow in the office we'll tag you guys for review when its ready.

**Comment by ardias** [comment]
Cool, thanks @Kav91

**Comment by haihongren** [comment]
> @haihongren Is this something that could not be accomplished with other functions? For example `split_objects`?
> Also, please add a test similar to what we have added recently for other functions.
> Thank you

@ardias Added the test for splitArrays function. thanks

File: internal/processor/flattener_test.go [hunk 1]
```diff
@@ -58,6 +59,31 @@ func TestSplitObjects(t *testing.T) {
 	}
 }
 
+func TestSplitArrays(t *testing.T) {
+	getAPI := func() *load.API {
+		return &load.API{
+			SplitObjects: true,
```
[the remark this answers is not in the document]
Comment (haihongren) [reply internal/processor/flattener_test.go]: Well spotted. split_objects is not required, removed from test now.
Also updated split_array in the docs/wiki/apis/functions.md

[the remark this answers is not in the document]
Comment (haihongren) [reply internal/processor/flattener_test.go]: @ardias 
I pulled the docs/wiki/apis/functions.md from master and updated it. 
But not sure why there is still conflict with docs/wiki/apis/functions.md. 
It looks the docs/wiki/apis/functions.md is being actively updated.  

How do you want me to update the docs/wiki/apis/functions.md with the new function?

**Pull request merged**